### ISSUE TITLE
Rumble stops working after a few uses with Shufflez's Xinput plugin

### ISSUE THREAD

**Issue by GoogleCodeExporter** [aloone/nulldc#27]
```
What steps will reproduce the problem?

1: Launch a game using this plugin and the rumble pack inserted. Observe 
rumble in game.
2: Close emulator and restart game

Result:
Rumble stops working entirely. This varies slightly, but after 3-4 games, 
it definetely stops functioning, until a computer restart is performed. 
This works absolutely fine when using the puru puru input plugin.

What version of the product are you using? What build? What plugins?
All versions that work with the Xinput controller plugin.

Default plugins for the following:
PowerVR Plugin: 
GDRom Plugin: 
AICA Plugin: 
ARM7 Plugin: 
Ext.Device Plugin: 

On what kind of system?
Win 7.

```

Original issue reported on code.google.com by `wespipe...@gmail.com` on 12 May 2010 at 6:10

**Comment by GoogleCodeExporter** [comment]
```
if you are using x360ce, it may be a bug in that.
```

Original comment by `danialho...@gmail.com` on 13 May 2010 at 12:07

**Comment by GoogleCodeExporter** [comment]
```
I thought that too, but the fact that it works fine for puru puru (and pretty 
much 
everything else I use it for) points to it being in issue with the plugin. BUT, 
if 
no one else can confirm this, that rumble works fine playing several games 
after one 
another, restarting the emu a few times, and it never stops then I guess we can 
throw this out. 

Please report your results with this plugin.
```

Original comment by `wespipe...@gmail.com` on 13 May 2010 at 4:25

**Comment by GoogleCodeExporter** [comment]
```
Please follow this thread:
http://www.xtemu.com/forum/nulldc-discussion-f170/nulldc-purupuru-xmaple-t3237.h
tml

Xmaple_Hangzhou, is a tryout for this issue. 
```

Original comment by `KrossX3` on 22 May 2010 at 10:44

Attachments:
- [Xmaple_Hangzhou.patch](https://storage.googleapis.com/google-code-attachments/nulldc/issue-27/comment-3/Xmaple_Hangzhou.patch)

**Comment by GoogleCodeExporter** [comment]
```
Its tested!
 XMaple_Hangzhou seems like no good.  XMaple_Hangzhou in SA2 vibro not working
absolutely---
Version XMaple_Groove3 better and gives me vibration. But in SA2 vibro seems 
like low
 (sometimes it seems is no vibration? but it still there only very low) (but in test
drive 6 and some other games much better I think normally), but in other games 
better. 
-- with XMaple_Groove3 much better!
something wrong with XMaple_Hangzhou
```

Original comment by `N2008eo...@rambler.ru` on 22 May 2010 at 11:35

**Comment by GoogleCodeExporter** [comment]
```
Does it still crash on exit or gets stuck with an effect?
```

Original comment by `KrossX3` on 23 May 2010 at 12:24

**Comment by GoogleCodeExporter** [comment]
```
It seems to be little to no change. It still crashes and gets stuck sometimes, 
but
Rez seems to have nicer effects.

This needs lots of works. >_<
```

Original comment by `KrossX3` on 23 May 2010 at 6:45

Attachments:
- [Xmaple_Hangzhou2.patch](https://storage.googleapis.com/google-code-attachments/nulldc/issue-27/comment-6/Xmaple_Hangzhou2.patch)

**Comment by GoogleCodeExporter** [comment]
```
Good luck then ;)
```

Original comment by `drkIIRaziel` on 23 May 2010 at 9:02

**Comment by GoogleCodeExporter** [comment]
```
Its tested!
Ok...
its better, I feel a  little more vibro in SA2, TD6, and like other games... 
but in
some games it still not working :( 
For Example: 18wheeler (I choose it---not works) and Crazy TAXI 1,2 (That game 
showed
me what vibropak not existing absolutely)

 P.s.
In future I think maybe somebody just create menu settings like NULLDC100b6 
(but just
a little better) Joystick plugin where we could:
---choose rumble level
---choose deadzone level
AND!
---customize any button! and dpad! (in Unreal Tournament for example 
alternative fire
- on dpad cant firing comfortly Bad marketing Sega you know. But if we have
"wonderful setting" we can change dpad on others buttons sometimes it very 
needful.
For example in Nulldc 100b6 I can do like that and its wonderful! ---
 By the way, now  we cant change dpad even in purupuru Joystick plugin) 


P.s.2
I know its fantastic tales, but this ideas so beutiful, isnt it?
```

Original comment by `N2008eo...@rambler.ru` on 23 May 2010 at 11:15

**Comment by GoogleCodeExporter** [comment]
```
I'll try to see why it crashes first though, then I'll check Crazy Taxi.

As for the rest, those are other issues. It's better to have one problem per 
issue,
easier to keep track of it. Also, perhaps some are related to Issue 35?

```

Original comment by `KrossX3` on 23 May 2010 at 11:25
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Alright!
I'll try put my "P.S." in new issue later:)
```

Original comment by `N2008eo...@rambler.ru` on 23 May 2010 at 12:19

**Comment by GoogleCodeExporter** [comment]
```
New one, although I haven't checked Crazy Taxi yet.
At least it seems to crash less often now. =_="

Anyway, three settings added to the INI:
Intensity ( % ) = To reduce or increase the rumble strength.
Length (ms) = Length of the effects.
RealFrequency [0,1] = When off, both motors have same effect.

Here's a patch, and there's a build in the usual place.
```

Original comment by `KrossX3` on 24 May 2010 at 1:01

Attachments:
- [Xmaple_Hangzhou3.patch](https://storage.googleapis.com/google-code-attachments/nulldc/issue-27/comment-11/Xmaple_Hangzhou3.patch)

**Comment by GoogleCodeExporter** [comment]
```
Just tested both Crazy Taxi and rumble is working.
```

Original comment by `KrossX3` on 24 May 2010 at 2:42

**Comment by GoogleCodeExporter** [comment]
```
Using the latest version everything works perfectly with purupuru and the 
Xinput 
controller plugins. Rumble always works properly on my games and the effects 
seem to 
work great now! 

Thanks for all your "input" to the input plugins. Get it. Sorry, nevermind. :) 
See 
ya!
```

Original comment by `wespipe...@gmail.com` on 24 May 2010 at 3:59

**Comment by GoogleCodeExporter** [comment]
```
There was still something more todo. Crazy Taxi didn't have engine rumble, but 
now it
has. ^_^

I think this is the one, with luck. =P


Here's the patch, and a build is already uploaded.
```

Original comment by `KrossX3` on 24 May 2010 at 7:35

Attachments:
- [Xmaple_Hangzhou4.patch](https://storage.googleapis.com/google-code-attachments/nulldc/issue-27/comment-14/Xmaple_Hangzhou4.patch)

**Comment by GoogleCodeExporter** [comment]
```
Its tested!
Much better CrossX3, Now I have vibro in 18wheeller! BUT not without bugs! 
In some games vibration works constantly! Vibration "non stop" with constantly 
power.
For example It happens in SonicAdventure1! Response is zzzzzz forever! 
Sometimes responce is not adecuate for actions on screen or with a little late 
in
some games. 
In Crazy Taxi vibration become better but then I stop my car I feel more 
vibration
than when I was trying to maximum press on the gas pedal! But maybe its just 
game
programming (they maybe wanted what we will feel power of car's motor?) And one 
no
good thing in Crazy Taxi vibro works only if I put "purupuru pack" in second 
slot,
and memorycard put in first slot!!!!!! If I replace it vibro will be 
disabled!!!! In
18wheeler too!!! But in other games it works in both variants!!!!
```

Original comment by `N2008eo...@rambler.ru` on 24 May 2010 at 8:45

**Comment by GoogleCodeExporter** [comment]
```
Alright... lets see.

So, constant rumble with Sonic Adventure 1... I don't have 18wheeler to test 
though.
Can you list the games with similar issues?

About Crazy Taxi, I think that's the way it's ment to be. And even then, there 
cannot
be more than one effect active. So at times, the engine rumble will shut off. I 
have
no clue if that's how it is on the real HW, but that's how it seems to be.

I'll check about the VMU <=> PuruPuru slot problem, but I think that should 
have its
own Issue.

Thanks a lot for your tests though. :)
```

Original comment by `KrossX3` on 24 May 2010 at 8:53

**Comment by GoogleCodeExporter** [comment]
```

List:

1) Constant zzzzzzz vibro problem:
Sonic Adventure 1 
San Francissco rush after first contact with opponent or wall or something
Fighting force 2
Rayman2


2) No vibration in any variants
headhunter
SPAWN
Legacy of Kain
Jet grind Radio
Record of Logoss war
Deadly Skies
```

Original comment by `N2008eo...@rambler.ru` on 24 May 2010 at 10:36

**Comment by GoogleCodeExporter** [comment]
```
Neat, thanks for the list.

As for those in #2. Are you sure those games do support rumble to begin with?
```

Original comment by `KrossX3` on 24 May 2010 at 12:03

**Comment by GoogleCodeExporter** [comment]
```
headhunter,Legacy of Kain, Record of Logoss war have 100% official vibro 
support just
look at game's menu!
about others ----- dont know, but sometimes support is exist. Only it 
automatic. Just
Like in Heavy Metal Geomatrix. Maybe and here something(specially SPAWN).....?



```

Original comment by `N2008eo...@rambler.ru` on 24 May 2010 at 12:15

**Comment by GoogleCodeExporter** [comment]
```
New tryout! Making the function anew, so please test. Specially the weird cases.

Xmaple_Hangzhou5 is uploaded already. 
No patch, cuz it's a mess of old and new stuff. >_<
```

Original comment by `KrossX3` on 25 May 2010 at 12:42

**Comment by GoogleCodeExporter** [comment]
```
Test Xmaple_Hangzhou6 instead. =P
```

Original comment by `KrossX3` on 25 May 2010 at 4:03

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Its tested! Much better! 
List of changes:
1)constan zzzzzzzzzz problem:
SA1 -- no ZZZZZ - OK!
San Francissco rush after first contact with opponent or wall or something --no 
zzz- OK!
Fighting force 2-- nozzz -Ok!
Rayman2 no zzzzzz--- but sometimes no response--- not Ok!


2) No vibration in any variants
headhunter -- Vibro works!
SPAWN -- vibro works only if purupurupack in 2nd slot
Legacy of Kain --- vibro works (only if purupurupack in 2nd slot)! but sometimes
zzzz--zzz(when i dont move) and
sometimes no responce
Jet grind Radio-- vibro works only if purupurupack in 2nd slot --- vibro not
allways!--- sometimes no responce --- 
Record of Logoss war --- no vibro
Deadly Skies --- no vibro

P.S. about 2nd slot just look at list in issue 76
THANKS!


```

Original comment by `N2008eo...@rambler.ru` on 25 May 2010 at 10:30

**Comment by GoogleCodeExporter** [comment]
```
Thanks, but please keep the "Slot problem" on issue 76. Although, I'll probably 
only
look into after this is commited.

Xmaple_Hangzhou7 is uploaded already. I've implemented the last section left, 
and
messed with the timmings again. See if there are any changes, your lists are 
quite
helpful.





```

Original comment by `KrossX3` on 25 May 2010 at 1:46

**Comment by GoogleCodeExporter** [comment]
```
Ooops, sorry. Messed up something. I'll reupload it later.
```

Original comment by `KrossX3` on 25 May 2010 at 2:02

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Tests of mine:
Record of Lodoss War: Working.
Jet Grind Radio: Working.
Rayman 2: Working. Might not trigger rumble somtimes, but seems to be the 
game's fault.

Could you test Soul Reaver about that "Zzzzz" stuck effect? If it still happens.
Could you upload the VMU (mediafire or something) and tell me the location 
please.
Also, from the cover, Deadly Skies doesn't seem to have rumble support.

About Settings:

Since the function was made anew, the effects are already scaled into the 
responsive
areas of the x360 controller. So I don't see much of a point for an intensity
setting. And timmings, are qute tight already. >_<

Might add them again later though. My focus now is getting rid of the bugs on 
it.
(The crash on exit is another story. >_< )
```

Original comment by `KrossX3` on 26 May 2010 at 12:43

**Comment by GoogleCodeExporter** [comment]
```
My test

Record of Lodoss War: Working.
Jet Grind Radio: Working, but very bad. 
Rayman 2: Working. Might not trigger rumble somtimes, but seems to be the game's
fault. very maybe

Soul Reaver. I think rumble effect is just game's specification. Maybe ZZZZ-ZZZZ
working adequately (my mess?) --- I mean when Raziel's energy going down and 
his body
covering electrical discharges I feel ZZZZ-ZZZZ, and When he dying and going to
underworld I feel constantly ZZZZZ(not all time just in situations Like that----
maybe it is my mess-- i'm sorry). Its normal responce, I think. When I'm 
attacking I
feel nothing, when enemies attack me I feel rumble. Maybe its normal 
responce......
About deadly skies I agree/ it seems has no rumble

abut vmu. I have many vmu file and where are many different saves becouse I 
need try
many games games when memory card in other slots. But I have better variant!!!!
I have VMU CD LOADER(All saves for All dc games) Better I think ask me what game
saves you need and I will create for you special VMU files with saves!!! How 
did you
think?
```

Original comment by `N2008eo...@rambler.ru` on 26 May 2010 at 1:25

**Comment by GoogleCodeExporter** [comment]
```
Hmm... it does seem like normal behavior in Soul Reaver. About Jet Grind Radio, 
why
do you think it's bad?

Rayman 2 has certain game bugs. For example, when you jump and use the helix to
glide; if you hold the button it won't rumble, but will if you just jump + jump 
it
will. That's a game's bug since no rumble info is sent.

Other games do have weird rumble choices. Like MvsC2, were it rumbles when 
you're hit
but not when you hit. Same goes for Cannon Spike. 


```

Original comment by `KrossX3` on 26 May 2010 at 1:33

**Comment by GoogleCodeExporter** [comment]
```
Do you have normal responce in Jet Grind Radio!? 
When I played this game I felt a rumble rarely, only once when jumping or 
something,
after that everything disappeared. After some time I felt  vibration was 
again...
once.... jump... run... no matter no responce... after few times I felt rumble 
again
just once .... after that i feel nothing... all starts again. Very bad! maybe 
game's
bug maybe not...
Anyway,...
here VMU Loader (All games)enjoy!!
http://www.mediafire.com/?zzyhmxqtejj  
```

Original comment by `N2008eo...@rambler.ru` on 26 May 2010 at 2:04

**Comment by GoogleCodeExporter** [comment]
```
From the tests I've done, Jet Grind Radio seems to rumble only when you bump 
into
something (when you hear the "ouch" or similar). And even then, it uses only 
mild/low
strength.

I guess I'll just add some settings and commit the thing. Since it appears to be
ready, or almost ready. =P

PS: Thanks for the VMU Loader. I'll give it a check later.
```

Original comment by `KrossX3` on 26 May 2010 at 2:59

**Comment by GoogleCodeExporter** [comment]
```
I FORGOT... Vmu better create on NULLDC1.0.3 or older version Nulldc!!!! After 
that
you must just copy "vmu_data_port" file into Nulldc1.0.4 becouse in nulldc1.0.4 
after
few minutes this disk becomes error!!!! I dont know why? But its easy 
procedure. 
```

Original comment by `N2008eo...@rambler.ru` on 26 May 2010 at 3:29

**Comment by GoogleCodeExporter** [comment]
```
Thanks, I'll keep it mind. (or probably just test the error in purporse =P)


Considered fixed by r29.
```

Original comment by `KrossX3` on 26 May 2010 at 3:46
- Changed state: **Fixed**
